GridList/GridList.jsx

```diff
@@ -0,0 +1,153 @@
+import { h } from "preact";
+import MaterialComponent from "../MaterialComponent";
+import Icon from "../Icon";
+
+const VALIDATION_VALUES_BY_KEY = {
+  "with-icon-align": ["start", "end"],
+  "tile-gutter": [1],
+  "tile-aspect-ratio": ["1x1", "16x9", "2x3", "3x2", "4x3", "3x4"]
+};
+
+const isValidValue = (validationValues, testValue) => {
+  return (
+    validationValues &&
+    validationValues.findIndex(val => val === testValue) >= 0
+  );
+};
+
+const notEmptyString = val => val !== "";
+
+/**
+ * @prop tile-gutter {1|4} - Number of px for the padding/spacing between items.
+ * @prop header-caption {boolean} - position <GridList.Tile.Secondary> at top
+ * @prop twoline-caption {boolean} - add spacing to <GridList.Tile.Secondary> for <GridList.Tile.SupportText>
+ * @prop with-icon-align {"start"|"end"} - position <GridList.Tile.Icon> at beginning or end of <GridList.Tile.Secondary>
+ * @prop tile-aspect-ratio {"1x1"|"16x9"|"2x3"|"3x2"|"4x3"|"3x4"} - aspect ratio for <GridList.Tile.Primary>
+ */
+class GridList extends MaterialComponent {
+  constructor() {
+    super();
+    this.componentName = "grid-list";
+    this._mdcProps = [
+      "header-caption",
+      "twoline-caption",
+      "tile-gutter",
+      "with-icon-align",
+      "tile-aspect-ratio"
```

Comment:
This would allow for invalid values, for example just writing ```<GridList tile-aspect-ratio>``` would add ```className="tile-aspect-ratio"``` on the element. Guess it's not that huge of an issue and it's a good idea to use ```_mdcProps``` to clean the element, so that's actually pretty nice.
Ping @prateekbh to see if we shouldn't add this pattern to ```MaterialComponent```:
```js
const VALIDATION_VALUES_BY_KEY = {
    "with-icon-align": ["start", "end"],
   "tile-gutter": [1],
   "tile-aspect-ratio": ["1x1", "16x9", "2x3", "3x2", "4x3", "3x4"]
 };
```

Reply:
this shouldn't really be here  since its a multi value prop.

also about clean up of props, i am starting to work on clean DOM PR soon, lets wait for it.